File: Spring-Boot/jpa-udemy/src/main/java/dev/vitorvidal/jpaudemy/entity/Student.java
package dev.vitorvidal.jpaudemy.entity;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
public class Student {
    @Id
    @GeneratedValue
    private Long id;
    @Column(nullable = false)
    private String name;
    @OneToOne(fetch = FetchType.LAZY)
    private Passport passport;

    @UpdateTimestamp
    private LocalDateTime lastUpdatedDate;
    @CreationTimestamp
    private LocalDateTime createdDate;

    protected Student() {
    }

    public Student(String name) {
        this.name = name;
    }

    public Passport getPassport() {
        return passport;
    }

    public void setPassport(Passport passport) {
        this.passport = passport;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Student{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: Spring-Boot/jpa-kotlin-example/src/main/kotlin/dev/vitorvidal/jpakotlinexample/entity/Student.kt
package dev.vitorvidal.jpakotlinexample.entity

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Student(
    @Id @GeneratedValue var id: Long? = null,

    @Column(nullable = false) var name: String? = null,

    @OneToOne(fetch = FetchType.LAZY) var passport: Passport? = null,

    @UpdateTimestamp val lastUpdatedDate: LocalDateTime? = null,

    @CreationTimestamp val createdDate: LocalDateTime? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        other as Student

        return id != null && id == other.id
    }

    override fun hashCode(): Int =
        this.id?.hashCode() ?: 0

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id )"
    }
}
